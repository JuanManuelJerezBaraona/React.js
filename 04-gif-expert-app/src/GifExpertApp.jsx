import { useState } from "react";
import { AddCategory, GifGrid } from "./components";


export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['dragon ball']);

    const onAddCategory = (newCategory) => {
        // console.log(newCategory)
        newCategory = newCategory.toLowerCase();

        if (categories.includes(newCategory)) return;

        setCategories([newCategory, ...categories]);
    }

    return (
        <>
            <h1>GifExpertApp</h1>

            <AddCategory 
                onNewCategory={value => onAddCategory(value)}
            />

            {
                categories.map(category => (
                    <GifGrid 
                        key={category} 
                        category={category} 
                    />
                ))
            }

        </>
    )
}
