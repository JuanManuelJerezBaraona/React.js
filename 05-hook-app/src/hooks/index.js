export * from './useFetch';
export * from './useTodo';