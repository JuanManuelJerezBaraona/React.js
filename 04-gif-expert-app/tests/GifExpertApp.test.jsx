import { render, screen } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";


describe('test on <GifExpertApp />', () => {

    test('should render the GifExpertApp component correctly', () => {

        render(<GifExpertApp />);
        // TODO:
        expect(screen.getByText('GifExpertApp')).toBeInTheDocument();

    });

    test('should display the initial category', () => {

        render(<GifExpertApp />);
        expect(screen.getByText('dragon ball')).toBeInTheDocument();
        
    });

});