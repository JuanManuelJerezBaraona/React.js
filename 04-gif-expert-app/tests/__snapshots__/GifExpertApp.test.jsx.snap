// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <GifExpertApp /> should render the GifExpertApp component correctly 1`] = `
<h1>
  GifExpertApp
</h1>
`;
